lib/stub_ui/lib/src/ui/text.dart

```diff
@@ -1117,6 +1296,13 @@ abstract class Paragraph {
   /// Coordinates of the [TextBox] are relative to the upper-left corner of the paragraph,
   /// where positive y values indicate down.
   List<TextBox> getBoxesForPlaceholders();
+
+  /// Returns the full list of [LineMetrics] that describe in detail the various
+  /// metrics of each laid out line.
+  ///
+  /// This can potentially return a large amount of data, so it is not recommended
+  /// to repeatedly call this. Instead, cache the results.
+  List<LineMetrics> getLineMetrics();
```

Comment:
These will need implementations in the _engine sub directory. In this case I would return an empty list similar to what the implementation for getBoxesForPlaceholders does.

cc @yjbanov

Reply:
Let me know if any of the metrics here are difficult to obtain for web. Some of them are quite libtxt specific.

cc @mdebbar

Reply:
I'm unlikely to get to it before @mdebbar is back. @mdebbar is currently redesigning out text layout. The new design will give us much more information, such as this. Our current design won't give us per-line information like this.